docs/salesforce.md

```diff
@@ -49,6 +49,179 @@ $ cci flow run test_everything
+name as the Contacts.
+
+```yaml
+# examples/salesforce/UsersForContacts.yml
+- plugin: snowfakery.standard_plugins.Salesforce.SalesforceQuery
+- plugin: snowfakery.standard_plugins.Salesforce
+
+- object: User
+  fields:
+    __random_contact:
+      SalesforceQuery.random_record:
+        from: Contact
+        fields: Id, FirstName, LastName
+
+    FirstName: ${{__random_contact.FirstName}}
+    LastName: ${{__random_contact.LastName}}
+    Alias: Grace
+    Username:
+      fake: Username
+    Email: ${{Username}}
+    TimeZoneSidKey: America/Bogota
+    LocaleSidKey: en_US
+    EmailEncodingKey: UTF-8
+    LanguageLocaleKey: en_US
+    ProfileId:
+      SalesforceQuery.find_record:
+        from: Profile
+        where: Name='Identity User'
+    # ContactId: ${{__random_contact.Id}}
+```
+
+In this case, the actual connection between the contact and
+the User is commented out because `Identity User` users cannot
+have Contacts, but you can see how you would connect a
+synthetic object to a pre-existing object, while also getting
+access to other fields.
+
+If you would like to use Salesforce query as a Dataset, that's
+another way that you can ensure that every synthetic record you
+create is associated with a distinct record from Salesforce.
+
+```yaml
+# examples/soql_dataset.yml
+- plugin: snowfakery.standard_plugins.Salesforce.SOQLDataset
+- object: Contact
+  count: 10
+  fields:
+    __users_from_salesforce:
+      SOQLDataset.iterate:
+        fields: Id, FirstName, LastName
+        from: User
+    OwnerId: ${{__users_from_salesforce.Id}}
+    FirstName: ${{__users_from_salesforce.FirstName}}
+    LastName: ${{__users_from_salesforce.LastName}}
+```
+
+Or if you'd like them in a random order, you can
+use `SOQLDataset.shuffle`:
+
+```yaml
+# examples/soql_dataset_shuffled.yml
+- plugin: snowfakery.standard_plugins.Salesforce.SOQLDataset
+- object: Contact
+  count: 10
+  fields:
+    __users_from_salesforce:
+      SOQLDataset.shuffle:
+        fields: Id, FirstName, LastName
+        from: User
+    # The next line depends on the users having particular
+    # permissions.
+    OwnerId: ${{__users_from_salesforce.Id}}
+    FirstName: ${{__users_from_salesforce.FirstName}}
+    LastName: ${{__users_from_salesforce.LastName}}
+```
+
+You may also specify a "where" clause to filter out irrelevant records:
+
+```yaml
+# examples/soql_dataset_where.yml
+- plugin: snowfakery.standard_plugins.Salesforce.SOQLDataset
+- object: Contact
+  count: 10
+  fields:
+    __users_from_salesforce:
+      SOQLDataset.shuffle:
+        fields: Id, FirstName, LastName
+        from: User
+        where: FirstName Like "A%"
+    OwnerId: ${{__users_from_salesforce.Id}}
+    FirstName: ${{__users_from_salesforce.FirstName}}
+    LastName: ${{__users_from_salesforce.LastName}}
+```
+
+### Testing Queries
+
+In general, you can test Snowfakery files outside of CumulusCI to see if they work:
```

Comment:
Would be helpful to have a sentence about what this does and doesn't do compared to the cci task.